test/new_relic/agent/instrumentation/rails/action_mailer_subscriber.rb

```diff
@@ -0,0 +1,189 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../../../test_helper'
+require 'new_relic/agent/instrumentation/action_mailer_subscriber'
+
+module NewRelic::Agent::Instrumentation
+  class TestMailer < ActionMailer::Base
+    SUBJECT = "Findings from Truart's estate"
+
+    default to: 'karras@mechanists.com',
+      from: 'garret@thecity.org',
+      reply_to: 'viktoria@constantine.net'
+
+    def test_action; end
+
+    def welcome
+      sleep until @_message.attachments
+      mail({subject: SUBJECT,
+            body: '<html><body>Regarding the mechanical servants...</body></html>'})
+    end
+  end
+
+  class ActionMailerSubscriberTest < Minitest::Test
+    ACTION = 'deliver'
+    NAME = "#{ACTION}.action_mailer"
+    ID = 1947
+    SUBSCRIBER = NewRelic::Agent::Instrumentation::ActionMailerSubscriber.new
+    MAILER = TestMailer.new
+
+    def setup
+      @delivery_method = ActionMailer::Base.delivery_method
+      ActionMailer::Base.delivery_method = :test
+    end
+
+    def teardown
+      ActionMailer::Base.delivery_method = @delivery_method
+    end
+
+    def test_start
+      in_transaction do |txn|
+        time = Time.now.to_f
+        SUBSCRIBER.start(NAME, ID, {mailer: MAILER.class.name})
+        segment = txn.segments.last
+
+        assert_in_delta time, segment.start_time
+        assert_equal "Ruby/ActionMailer/#{MAILER.class.name}/#{ACTION}", segment.name
+      end
+    end
+
+    def test_start_when_not_traced
+      SUBSCRIBER.state.stub :is_execution_traced?, false do
+        in_transaction do |txn|
+          SUBSCRIBER.start(NAME, ID, {})
+
+          assert_empty txn.segments
+        end
+      end
+    end
+
+    def test_start_with_exception_raised
+      logger = MiniTest::Mock.new
+
+      NewRelic::Agent.stub :logger, logger do
+        logger.expect :error, nil, [/Error during .* callback/]
+        logger.expect :log_exception, nil, [:error, ArgumentError]
+
+        in_transaction do |txn|
+          SUBSCRIBER.stub :start_segment, -> { raise 'kaboom' } do
+            SUBSCRIBER.start(NAME, ID, {})
+          end
+
+          assert_equal 1, txn.segments.size
+        end
+      end
+      logger.verify
+    end
+
+    # def test_finish_with_exception_raised
+    #   logger = MiniTest::Mock.new
+
+    #   NewRelic::Agent.stub :logger, logger do
+    #     logger.expect :error, nil, [/Error during .* callback/]
+    #     logger.expect :log_exception, nil, [:error, ArgumentError]
+
+    #     in_transaction do |txn|
+    #       SUBSCRIBER.stub :pop_segment, -> { raise 'kaboom' } do
+    #         SUBSCRIBER.finish(NAME, ID, {})
+    #       end
+
+    #       assert_equal 1, txn.segments.size
+    #     end
+    #   end
+    #   logger.verify
+    # end
```

Comment:
Did you mean to leave this commented?

Reply:
I could have sworn I deleted it multiple times already. Weird. It is a slightly different variation on a test of the same name that is not commented out. Hopefully finally removed now with [e71e37b](https://github.com/newrelic/newrelic-ruby-agent/pull/1740/commits/e71e37bde75ea6622e789faeecf8c8976a66dd53)